From eec95e69e5d0dcc16fd8872c1ad680265a96cce3 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Sat, 13 Jun 2020 17:44:45 +0200
Subject: [PATCH] dlmisc.c: fix uclibc build

RTLD_DL_LINKMAP & RTLD_DI_ORIGIN, are unsupported on uClibc:
https://cgit.uclibc-ng.org/cgi/cgit/uclibc-ng.git/tree/include/dlfcn.h#n106

This patch adds detection for uClibc because uClibc also defines
__GLIBC__:
https://lists.gnu.org/archive/html/bug-gnulib/2010-11/msg00280.html

BugLink: https://github.com/alsa-project/alsa-lib/pull/58
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
Backported from upstream eec95e69e5d0dcc16fd8872c1ad680265a96cce3
---
 src/dlmisc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dlmisc.c b/src/dlmisc.c
index 5d672146..ca94b123 100644
--- a/src/dlmisc.c
+++ b/src/dlmisc.c
@@ -69,7 +69,7 @@ static inline void snd_dlpath_unlock(void) {}
 int snd_dlpath(char *path, size_t path_len, const char *name)
 {
 #ifdef HAVE_LIBDL
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
 	static int plugin_dir_set = 0;
 	snd_dlpath_lock();
 	if (!plugin_dir_set) {
-- 
2.20.1

